docs/content/release_notes/limitations.md

```diff
@@ -6,35 +6,84 @@ As opposed to known issues, limitations are functionality restrictions that are
+### HyperSwap volume limitations
+The following IBM block storage CSI driver features are not supported on volumes where HyperSwap is used:
 
--   When deleting a PersistentVolumeClaim (PVC), the persistent volume (PV) remains until all snapshots of the specific PV are deleted.
--   When using the CSI (Container Storage Interface) driver with IBM Spectrum® Virtualize family products, a snapshot can only be used to provision a new volume of equal size.
+- A HyperSwap volume cannot be created from a snapshot.
 
-**Note:** For volume snapshot limitations pertaining specifically to HyperSwap usage, see [HyperSwap volume limitations](#hyperswap-volume-limitations).
+    **Note:** A snapshot can be created from a HyperSwap volume.
+- Volume cloning.
+
+### Stretched volume limitations
+ When conducting volume cloning, both volumes must use stretched topology.
+
+## I/O group limitations
+
+I/O group configuration is only supported for use with IBM Spectrum Virtualize family storage systems.
+
+## NVMe®/FC usage limitations
+
+ Red Hat® Enterprise Linux CoreOS (RHCOS) does not support NVMe®/FC.
+ 
+ For other limitations with your storage system, see the following section within your Spectrum Virtualize product documentation on [IBM Documentation](https://www.ibm.com/docs/en/): **Configuring** > **Host attachment** > **NVMe over Fibre Channel host attachments** > **FC-NVMe limitations and SAN configuration guidelines**.
+
+## Snapshot function limitations
+
+**Important:** Snapshot function support is only Alpha support.
+
+- Snapshot function is only supported for use with IBM Spectrum Virtualize family storage system versions 8.5.1 or higher. For more information, see **Product overview** > **Technical overview** > **Volume groups** > **Snapshot function** within your Spectrum Virtualize product documentation on [IBM Documentation](https://www.ibm.com/docs).
+- Both source and target PVCs (in a source PVC to snapshot to target PVC scenario) must have the same space efficiency set within their storage classes. If the space efficiency is set differently, the target PVC creation fails.
+- A PVC target must have the same volume size as the source volume.
+- A snapshot that uses the Snapshot function cannot be created with space efficiency set. If the VolumeSnapshotClass has the `SpaceEfficiency` parameter set along with the snapshot flag (`virt_snap_func`) enabled, the snapshot creation fails.
+- In very rare cases, there can be leftover or undeleted volumes. As a result of the Kubernetes/Openshift and CSI being stateless, in cases where the storage is not able to save a specific state, the CSI driver might administer the wrong process.
+
+    For example, this can happen in a case where a volume is created from a snapshot and then a request to delete the same volume. In such a case, the driver might delete a volume that was not created from the snapshot. This occurs as a result of no record that is kept if the volume is created from a snapshot or not.
```

Comment:
we won't delete wrong volumes, they have strong identifiers.
we might however delete the wrong snapshots - if a deletion process of a certain snapshot is finished on the storage but the driver crush and didn't register that, if another snapshot is created before the retry it will take the deleted snapshot id and we would delete the new snapshot when retry to delete the old one.

hope it's clear

Reply:
this is for the idempotency?
@erantzabari said that we shouldn't be mentioning the UID issue.

Reply:
updated per our conversation @ArbelNathan  please approve. thanks